### ISSUE TITLE
Upgrade redux-saga to 1.0

### ISSUE THREAD

**Issue by charl3sj** [marmelab/react-admin#3029]
**What you were expecting:**
Include react-admin into existing project. Followed instructions on: https://marmelab.com/react-admin/CustomApp.html

**What happened instead:**
Got
```
TypeError: effectRunner is not a function
    at runEffect (redux-saga-core.esm.js:1210)
    at digestEffect (redux-saga-core.esm.js:1277)
    at next (redux-saga-core.esm.js:1167)
    at proc (redux-saga-core.esm.js:1114)
    at eval (redux-saga-core.esm.js:595)
    at immediately (redux-saga-core.esm.js:56)
    at runForkEffect (redux-saga-core.esm.js:594)
    at runEffect (redux-saga-core.esm.js:1210)
    at digestEffect (redux-saga-core.esm.js:1277)
    at eval (redux-saga-core.esm.js:683)
    at Array.forEach (<anonymous>)
    at runAllEffect (redux-saga-core.esm.js:682)
    at runEffect (redux-saga-core.esm.js:1210)
    at digestEffect (redux-saga-core.esm.js:1277)
    at next (redux-saga-core.esm.js:1167)
    at proc (redux-saga-core.esm.js:1114)
    at eval (redux-saga-core.esm.js:1374)
    at immediately (redux-saga-core.esm.js:56)
    at runSaga (redux-saga-core.esm.js:1373)
    at Function.sagaMiddleware.run (redux-saga-core.esm.js:1434)
    at configureStore (store.js:28)
    at eval (store.js:33)
    at Module../src/store.js (main.chunk.js:270)
    at __webpack_require__ (bundle.js:782)
    at fn (bundle.js:150)
    at eval (index.js:11)
    at Module../src/index.js (main.chunk.js:187)
    at __webpack_require__ (bundle.js:782)
    at fn (bundle.js:150)
    at Object.0 (main.chunk.js:282)
    at __webpack_require__ (bundle.js:782)
    at checkDeferredModules (bundle.js:46)
    at Array.webpackJsonpCallback [as push] (bundle.js:33)
    at main.chunk.js:1
```

**Steps to reproduce:**
_rootSaga.js_
```
import { all, fork } from 'redux-saga/effects';
import { adminSaga, defaultI18nProvider } from 'react-admin';
import jsonServerProvider from 'ra-data-json-server';
import { initialiseCognito, userInfoWatcher } from "./app/saga";

export const authProvider = () => Promise.resolve();
export const dataProvider = jsonServerProvider('http://jsonplaceholder.typicode.com');
const i18nProvider = defaultI18nProvider;

export default function* rootSaga() {
    yield all([
        adminSaga(dataProvider, authProvider, i18nProvider),
        initialiseCognito,
        userInfoWatcher
    ].map(fork));
}
```

_store.js_
```
import { applyMiddleware, compose, createStore } from "redux";
import createSagaMiddleware from 'redux-saga';
import { formMiddleware } from 'react-admin';
import { routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createHashHistory';
import rootReducer from "./rootReducer";
import rootSaga from "./rootSaga";

export const adminHistory = createHistory();

const configureStore = initialState => {
    const sagaMiddleware = createSagaMiddleware();

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

    const store = createStore(rootReducer,
        initialState,
        composeEnhancers(
            applyMiddleware(
                sagaMiddleware,
                formMiddleware,
                routerMiddleware(adminHistory),
            )
        )
    );
    sagaMiddleware.run(rootSaga);
    return store;
};

export const store = configureStore();
```

_index.js_
```
import { store } from './store';
ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
         <App /> 
      </BrowserRouter>
    </Provider>,
    document.getElementById('root')
);
```

_OrgAdmin.js_
```
import React, { Component } from "react";
import PropTypes from 'prop-types';
import { Admin, Resource, ListGuesser } from "react-admin";
import { authProvider } from "../rootSaga";
import { store, adminHistory } from "../store";

export class OrgManager extends Component {
    getChildContext() {
        return { store }
    }

    render() {
        return (
            <Admin authProvider={authProvider} history={adminHistory} title="Manage Organisation">
                <Resource name="users" list={ListGuesser} />
            </Admin>
        );
    }
}

OrgManager.childContextTypes = {
    store: PropTypes.object
};
```

**Related code:**
Available here: https://github.com/OpenCHS/openchs-webapp/tree/exp/src

**Other information:**
StackOverflow link with screenshots: https://stackoverflow.com/questions/55236556/in-custom-app-with-embedded-admin-adminsaga-doesnt-work

**Environment**

* React-admin version: 2.7.2
* React version: 16.8.3
* Redux version: 4.0.1
* Redux-saga version: 1.0.2 
* Browser: Chrome 72.0.x

**Comment by Kmaschta** [comment]
Mind you fork [this CodeSandbox](https://codesandbox.io/s/349ovjmlx5) and reproduce the error in it? This will help maintainers to find the issue quickly

**Comment by charl3sj** [comment]
Done! Find the CodeSandbox here: https://codesandbox.io/s/znm7369zk4

I followed a lead from https://github.com/redux-saga/redux-saga/issues/1728 and downgraded redux-saga to 0.16.2 (which is what RA uses internally) and then it works! See https://codesandbox.io/s/23o0m0kmxn

**Comment by fzaninotto** [comment]
React-admin correctly specifies a dependency on redux-saga "~0.16.0". If you use redux-saga 1.0 in your custom app, that means that 2 different versions of saga are installed.

I'm not sure we can do anything more about your bug - the problem is that redux-saga cannot run two different versions in parallel.

In the long run, we'll have to upgrade redux-saga to 1.0, but that's a breaking change for existing applications, so it won't happen before 3.0. I'm modifying the title of this issue to reflect that.

**Comment by aislanmaia** [comment]
Things like that is what blocking me to use this library (`react-admin`), because I want to use goodness from more latest versions of libs like Material-ui and so on. And the `react-admin` seems pretty behindhand.

**Comment by djhi** [comment]
You can always fork the library

**Comment by macrozone** [comment]
i think the jump from 0.16.0 to 1.0.0 has no relevant breaking change. I think more people struggle with downgrading their apps to 0.16.0 than are "forced" to upgrade saga to 1.0.0.

Similar situation with material-ui. react-admin works with 3.x.x so far (overwriting the version with resolutions). But its just additional headache and you will never be sure if it works properly...

**Comment by fzaninotto** [comment]
implemented in #3212
